lib/QoreLib.cpp

```diff
@@ -2664,3 +2672,57 @@ QoreHashNode* get_source_location(const QoreProgramLocation* loc) {
 const char* type_get_name(const QoreTypeInfo* t) {
     return QoreTypeInfo::getName(t);
 }
+
+void qore_delete_module_options() {
+    if (mod_opt_map.empty()) {
+        return;
+    }
+    ExceptionSink xsink;
+    for (auto& i : mod_opt_map) {
+        for (auto& vi : i.second) {
+            vi.second.discard(&xsink);
+        }
+    }
+}
+
+void qore_set_module_option(const char* mod, const char* opt, QoreValue val) {
+    AutoLocker al(mod_opt_lock);
+    mod_opt_val_map_t::iterator vi;
+    mod_opt_map_t::iterator i = mod_opt_map.lower_bound(mod);
+    if (i == mod_opt_map.end() || i->first != mod) {
```

Comment:
good catch!

Reply:
updated

Reply:
when we move to c++17 we can use [string_view](https://stackoverflow.com/questions/20803826/what-is-string-view) in such cases to avoid unnecessary allocations

Reply:
ok done